src/averageAges.js

```diff
@@ -15,11 +16,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const persons = century
+    ? people.filter(person => century === Math.ceil(person.died / 100) && person.sex === 'm')
+    : people.filter(person => person.sex === 'm');
+
+  const allAges = persons.map(person => person.died - person.born);
+  const average = (allAges.reduce((accumulator, personsAge) => accumulator + personsAge) / allAges.length).toFixed(2);
+
+  return Number(average);
```

Comment:
move to the function helper
```suggestion
  const average = getAverageAge(allAges);
  
  return average;
```